python/cuspatial/cuspatial/core/_column/geocolumn.py

```diff
@@ -30,23 +31,118 @@ def __init__(
+            Rings: List of polygon ring offsets
+            Polygons: Offset into rings of each polygon
+
+            Finally, each set of offsets must have the length of the
+            array appended to the end, as Arrow offset lists are length
+            n + 1 but our original shapefile code offset lists are only
+            length n.
+            """
+            polygons_col = data[0].astype("int32")
+            rings_col = data[1].astype("int32")
+            coordinates = (
+                data[2].stack().astype("float64").reset_index(drop=True)
+            )
+            """
+            Store a fixed-size offsets buffer of even numbers:
+            0   0
+            1   2
+            2   4
+            ...
+            Up to the size of the original input.
+            """
+            coordinate_offsets = cudf.concat(
+                [
+                    cudf.Series(
+                        cp.arange(len(coordinates) // 2) * 2, dtype="int32"
+                    ),
+                    cudf.Series([len(coordinates) // 2 * 2], dtype="int32"),
+                ]
+            ).reset_index(drop=True)
+            """
+            Store a points offset buffer: 0 - n
+            """
+            point_offsets = cudf.concat(
+                [
+                    cudf.Series(
+                        cp.arange(len(coordinates) // 2), dtype="int32"
+                    ),
+                    cudf.Series([len(coordinates) // 2 - 1], dtype="int32"),
+                ]
+            ).reset_index(drop=True)
+            rings_offsets = cudf.concat(
+                [
+                    cudf.Series(rings_col),
+                    cudf.Series([len(point_offsets) - 1], dtype="int32"),
+                ]
+            ).reset_index(drop=True)
+            polygons_offsets = cudf.concat(
+                [
+                    cudf.Series(polygons_col),
+                    cudf.Series([len(polygons_col)], dtype="int32"),
+                ]
+            ).reset_index(drop=True)
+            coords = cudf.core.column.ListColumn(
+                size=len(coordinate_offsets) - 1,
+                dtype=cudf.ListDtype(coordinates.dtype),
+                children=(coordinate_offsets._column, coordinates._column),
+            )
+            points = cudf.core.column.ListColumn(
+                size=len(rings_offsets) - 1,
+                dtype=cudf.ListDtype(coords.dtype),
+                children=(rings_offsets._column, coords),
+            )
+            rings = cudf.core.column.ListColumn(
+                size=len(polygons_offsets) - 1,
+                dtype=cudf.ListDtype(points.dtype),
+                children=(polygons_offsets._column, points),
+            )
+            parent = cudf.core.column.ListColumn(
+                size=len(polygons_offsets) - 1,
+                dtype=cudf.ListDtype(rings.dtype),
+                children=(polygons_offsets._column, rings),
+            )
```

Comment:
I think `points` ListColumn is not needed. The number of rings should equal to the length of `ring_offsets - 1`, not `polygon_offsets-1`.

```suggestion
            coords = cudf.core.column.ListColumn(
                size=len(coordinate_offsets) - 1,
                dtype=cudf.ListDtype(coordinates.dtype),
                children=(coordinate_offsets._column, coordinates._column),
            )
            rings = cudf.core.column.ListColumn(
                size=len(rings_offsets) - 1,
                dtype=cudf.ListDtype(points.dtype),
                children=(rings_offsets._column, coords),
            )
            parent = cudf.core.column.ListColumn(
                size=len(polygons_offsets) - 1,
                dtype=cudf.ListDtype(rings.dtype),
                children=(polygons_offsets._column, rings),
            )
```

Reply:
I would prefer this, but it simply doesn't produce the same data structure as what is made in `from_geopandas`. Maybe there is a bug there...

Reply:
The bug had to do with my naming scheme, which was confused. In the latest PR the structures are now named appropriately for what they represent. Thanks for noticing!